src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 16;
+    static final double DEFAULT_LOAD_FACTOR = 0.75;
+    static final int CAPACITY_MODIFIER = 2;
+    private int size;
+    private Node<K, V>[] table;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        int index = Math.abs(hash) % table.length;
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+            newNode.next = table[index];
+            table[index] = newNode;
+        }
+        size++;
+        if (size == threshold) {
```

Comment:
let's check the value before adding the element, because you will be adding the same element twice

Reply:
not fixed